homeassistant/components/media_player/webostv.py

```diff
@@ -343,6 +343,39 @@ def select_source(self, source):
             self._current_source = source_dict['label']
             self._client.set_input(source_dict['id'])
 
+    def play_media(self, media_type, media_id, **kwargs):
+        """Find app for media_id and initialize with url"""
+        youtube_id = self.youtube_video_id(media_id)
```

Comment:
In this function. Just check `if media_type == "youtube"`

Reply:
We should remove all YouTube url parsing. The user will need to pass in the ID themselves

Reply:
No no no no, I have to stand against this. The correct approach would be to create new class "MediaTypes.Youtube" or something like that and insert the parsing here. The parsing is required. I did not know where to put it, so I put it here. Why? Because it is tied for this particular case, but it is true that it might be valuable also for someone else. Why is it required? Because you see only part of code. This was my first pull request just to try how it works here. It is required in order to give voice commands to TV like "Play {event or artist or video name} on Youtube" intent. Then it will correctly search Youtube and return the id. You don't want to let user tell a voice command: "Play youtube id" and then spell hex encoded characters lol.

Reply:
So it is not just some fancy thing to let the core user writing scripts to make life easier. It is actually functional code for your voice assistant. My voice assistant can use your intents, but it is also multilanguage and supports TTS out of the box. You speak vietnamese, you get vietnamese response. You speak english, you get english response. You speak czech, you get czech response. All for the same command with simple things. Searching Youtube & Spotify for music or content is basic feature that should be supported.

Reply:
Sir, I don't want to be rude. But did I tell that any of these use cases are part of LG platform? NO, big no again.

Reply:
Ok, so tell me where to put Youtube parsing method in a way that it can be used as part of intent.

Reply:
I did only outline my whole solution that is split across several source files. Of course this functionality is not part of LG source code.

Reply:
This conversation is stupid and it only takes away my time. First time when u write: "Remove all the YouTube url parsing from this PR, that does not belong in Home Assistant." ...you should write where do you want to put the URL parsing method. Because then I had to explain the whole user story and u did not get it for 2nd time :-D ...sorry, no offense, just facts.

Reply:
Somewhere must be defined route: intent -> keyword -> search youtube for keyword -> pass id to LG. I'm interested how this would be coded on your current platform in a such way that user can replace LG implementation with other one.

Reply:
You have to add Youtube parsing to your own intent handler.

Reply:
I don't feel like arguing about these things anymore. My decision is final. If you don't want to make the necessary changes, close this PR as it will not be merged.

Reply:
Okay, thank you. I will update this PR. Where to put Youtube parsing please? How to make it best experience for users, so they don't have to write their own macro or script?

Reply:
You know if you would help right away, we both save some time. Now it's your turn to show you are not what I think you are :-)